### ISSUE TITLE
MRTK vNext SpeechInputHandler NullReference

### ISSUE THREAD

**Issue by alcooper91** [Microsoft/MixedRealityToolkit-Unity#2834]
I'm attempting to utilize MRTK vNext to add some speech commands to an app, but I'm hitting some issues editing my scene and then had a question about usage.  Sorry if this isn't the right place for posting, but I didn't see any docs around what I'm attempting either.

Scenario:
I'm attempting to add speech to my app, I created a Custom MixedRealityConfigurationProfile that matches the default except for a custom InputSystemProfile, that matches the default except for a custom SpeechProfile to add a few voice commands.  When I first add new VoiceCommands, I'm able to interact with the SpeechInputHandler to register new commands; but when I go back in and edit the script/functions that are being targeted by that command, I end up being told that the Handler thinks there are no keywords registered, and viewing the Handler causes the following NullReference to appear in the output:
NullReferenceException: Unable to find IMixedRealitySpeechSystem.
Microsoft.MixedReality.Toolkit.Core.Managers.MixedRealityManager.GetManager (System.Type type) (at Assets/MixedRealityToolkit/_Core/Managers/MixedRealityManager.cs:502)
Microsoft.MixedReality.Toolkit.Core.Managers.MixedRealityManager.GetManager[T] () (at Assets/MixedRealityToolkit/_Core/Managers/MixedRealityManager.cs:473)
Microsoft.MixedReality.Toolkit.Core.Definitions.InputSystem.MixedRealityInputSystemProfile.get_SpeechSystem () (at <5633ad60a2174c55802b45e61d12df29>:0)
Microsoft.MixedReality.Toolkit.Core.Definitions.InputSystem.MixedRealityInputSystemProfile.get_IsSpeechCommandsEnabled () (at <5633ad60a2174c55802b45e61d12df29>:0)
Microsoft.MixedReality.Toolkit.SDK.Inspectors.Input.Handlers.SpeechInputHandlerInspector+<RegisteredKeywords>c__Iterator0.MoveNext () (at Assets/MixedRealityToolkit-SDK/Inspectors/Input/Handlers/SpeechInputHandlerInspector.cs:141)
System.Linq.Set`1[TElement].UnionWith (System.Collections.Generic.IEnumerable`1[T] other) (at <839a3cb835c04d14aeb58d83bb7bc4bd>:0)
System.Linq.Enumerable+DistinctIterator`1[TSource].FillSet () (at <839a3cb835c04d14aeb58d83bb7bc4bd>:0)
System.Linq.Enumerable+DistinctIterator`1[TSource].ToArray () (at <839a3cb835c04d14aeb58d83bb7bc4bd>:0)
System.Linq.Enumerable.ToArray[TSource] (System.Collections.Generic.IEnumerable`1[T] source) (at <839a3cb835c04d14aeb58d83bb7bc4bd>:0)
Microsoft.MixedReality.Toolkit.SDK.Inspectors.Input.Handlers.SpeechInputHandlerInspector.OnEnable () (at Assets/MixedRealityToolkit-SDK/Inspectors/Input/Handlers/SpeechInputHandlerInspector.cs:31)

Unmodified/previously hooked up commands are fine, and when I press play in the editor I can see all of the keywords again, but as soon as I stop playing it reverts to the error message about no keywords.

**Comment by StephenHodgson** [comment]
@alcooper91 Thanks for bringing this to our attention.

Can you give me some solid repro steps to fix this please?

**Comment by StephenHodgson** [comment]
Looks like the speech system is missing form the component registry. Can you double check to make sure it's there?

![image](https://user-images.githubusercontent.com/13334553/46036892-8eae3900-c0bb-11e8-859a-2012eaf7d4b7.png)

**Comment by alcooper91** [comment]
Hi Stephen,
Yes I still have the DefaultMixedRealityRegisteredComponentsProfile added.  The only thing I've customized is the SpeechProfile to add a few phrases.
I then added a SpeechInputHandler on the MixedRealityManager, unchecked "Require Focus" and then targeted a script on another GameObject.  Things seem fine until I make a change to that script that is being targeted by the SpeechInputHandler, whereupon I get the NullReference and the SpeechInputHandler thinking there are no keywords.
However, when I press "Play" everything seems to be working.

**Comment by StephenHodgson** [comment]
>I then added a `SpeechInputHandler` on the `MixedRealityManager`,

Yeah I bet that's the problem. You might want to put that bad boy on it's own game object

**Comment by StephenHodgson** [comment]
Okay thanks for the write up. I'll look into this asap

**Comment by StephenHodgson** [comment]
Okay, I can't seem to reproduce your error with the latest development branch.

1. I removed then created my own speech command profile
2. Created two speech commands: Hello and Goodbye
3. Added a speech input handler in the scene and set both hello and goodbye unity events.
4. Pressed play with no errors.

**Comment by StephenHodgson** [comment]
Is there something else you're doing that may be causing the error to manifest?

**Comment by StephenHodgson** [comment]
Got it.

**Comment by StephenHodgson** [comment]
Seems as though the errors crop up when the handler is trying to initialize the manager. 

Doesn't actually break anything, but we should be handling them better. I'll see what can be done.

**Comment by StephenHodgson** [comment]
Still can't repro the original issue with eh custom speech command profile tho.

**Comment by alcooper91** [comment]
Given the information from Kurtis I expect that the custom profile may not be necessary.

**Comment by alcooper91** [comment]
However, I do see everything work temporarily until a reboot of the editor.

**Comment by StephenHodgson** [comment]
I was able to repro this when working on platforms that were not set to windows or windows universal.
